modules/iam/data.tf

```
@@ -2,7 +2,444 @@ data "aws_caller_identity" "current" {}
+    ]
+  }
+
+  statement {
+    sid       = "KMSforSNS"
+    effect    = "Allow"
+    resources = ["${var.kms_arn}"]
+
+    actions = [
+      "kms:Decrypt",
+      "kms:Encrypt",
+      "kms:GenerateDataKey",
+    ]
+  }
+
+  statement {
+    sid       = "DynamoDB"
+    effect    = "Allow"
+    resources = ["${local.ddb_table_arn}"]
+
+    actions = [
+      "dynamodb:PutItem",
+      "dynamodb:Query",
+      "dynamodb:Scan",
+      "dynamodb:UpdateItem",
+    ]
+  }
+
+  statement {
+    sid       = "StartStateMachine"
+    effect    = "Allow"
+    resources = ["${var.state_machine_arn}"]
+    actions   = ["states:StartExecution"]
+  }
+}
+
+data "aws_iam_policy_document" "basic" {
+  statement {
+    sid       = "WriteToCloudWatchLogs"
+    effect    = "Allow"
+    resources = ["arn:aws:logs:*:*:*"]
+
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+    ]
+  }
+
+  statement {
+    sid       = "PutCloudWatchMetrics"
+    effect    = "Allow"
+    resources = ["*"]
+    actions   = ["cloudwatch:PutMetricData"]
+  }
+
+  statement {
+    sid    = "SNS"
+    effect = "Allow"
+
+    resources = [
+      "arn:aws:sns:*:*:${var.project}-${var.environment}",
+      "arn:aws:sns:*:*:${var.project}-dlq-${var.environment}",
+    ]
+
+    actions = [
+      "sns:Publish",
+      "sns:Subscribe",
+    ]
+  }
+
+  statement {
+    sid       = "KMSforSNS"
+    effect    = "Allow"
+    resources = ["${var.kms_arn}"]
+
+    actions = [
+      "kms:Decrypt",
+      "kms:Encrypt",
+      "kms:GenerateDataKey",
+    ]
+  }
+
+  statement {
+    sid       = "GetAccountName"
+    effect    = "Allow"
+    resources = ["*"]
+    actions   = ["iam:ListAccountAliases"]
+  }
+}
+
+data "aws_iam_policy_document" "states" {
+  statement {
+    sid       = "CloudWatchLogs"
+    effect    = "Allow"
+    resources = ["*"]
+
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+      "logs:CreateLogDelivery",
+      "logs:GetLogDelivery",
+      "logs:UpdateLogDelivery",
+      "logs:DeleteLogDelivery",
+      "logs:ListLogDeliveries",
+      "logs:PutResourcePolicy",
+      "logs:DescribeResourcePolicies",
+      "logs:DescribeLogGroups",
+    ]
+  }
+
+  statement {
+    sid       = "XRay"
+    effect    = "Allow"
+    resources = ["*"]
+
+    actions = [
+      "xray:PutTraceSegments",
+      "xray:PutTelemetryRecords",
+      "xray:GetSamplingRules",
+      "xray:GetSamplingTargets",
+    ]
+  }
+
+  statement {
+    sid       = "PutCloudWatchMetrics"
+    effect    = "Allow"
+    resources = ["*"]
+    actions   = ["cloudwatch:PutMetricData"]
+  }
+
+  statement {
+    sid       = "Lambda"
+    effect    = "Allow"
+    resources = ["arn:aws:lambda:*:*:function:${var.project}-*-${var.environment}"]
+    actions   = ["lambda:InvokeFunction"]
+  }
+
+  statement {
+    sid       = "KMSforSNS"
+    effect    = "Allow"
+    resources = ["${var.kms_arn}"]
```

nit: can we change these to this?

```suggestion
    resources = [var.kms_arn]
```